src/Lecture2.hs

```diff
@@ -155,12 +178,70 @@ You're free to define any helper functions.
 
 -- some help in the beginning ;)
 data Knight = Knight
-    { knightHealth    :: Int
-    , knightAttack    :: Int
-    , knightEndurance :: Int
-    }
-
-dragonFight = error "TODO"
+  { knightHealth    :: Int
+  , knightAttack    :: Int
+  , knightEndurance :: Int
+  }
+
+knightTurn :: (Knight, Dragon) -> (Knight, Dragon)
+knightTurn (knight, dragon) =
+  (knight {knightEndurance = (knightEndurance knight) - 1}
+  ,dragon {dragonHealth = (dragonHealth dragon) - (knightAttack knight)})
+
+data Treasure = Treasure
```

Comment:
That's a nice trick to solve the most difficult part of this task 😅 
If it was some `Treasure a` and not just `Treasure`, the solution will be slightly more challenging 😉

Reply:
Oh, did I cheat myself out of a lesson? Whoops! Maybe I'll revisit it in the future...